src/views/CountDetail.vue

```diff
@@ -290,10 +345,43 @@ onIonViewDidEnter(async() => {
   selectedSegment.value = 'all';
   queryString.value = '';
   updateFilteredItems();
+  previousItem = itemsList.value[0]
   await store.dispatch("product/currentProduct", itemsList.value[0])
   updateNavigationState(0);
 })  
 
+async function focusInput() {
+  barcodeInput.value.focus();
+  isInputFocused.value = true;
+}
+
+function handleBlur() {
+  isInputFocused.value = false;
+}
+
+async function handleInput(event) {
+  if (!isInputFocused.value) return; 
+  let sku = event.target.value;
+  const cachedProducts = getCachedProducts.value;
+  let scannedItem;
+
+  scannedItem = Object.keys(cachedProducts).find(productId => cachedProducts[productId].sku === sku);
+  if (!scannedItem) {
+    scannedItem = await findProduct(sku);
```

Comment:
Check this flow once, I think this will fail after findProduct method is called, as we are returning the resp from findProduct and not actual product so the check on line 373, will always be false.